extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -345,6 +347,44 @@ export class AmpA4A extends AMP.BaseElement {
 
     /** @private {string} */
     this.safeframeVersion_ = DEFAULT_SAFEFRAME_VERSION;
+
+    /** @private @const {!./refresh-manager.RefreshManager}  */
+    this.refreshManager_ = refreshManagerFor(this.win);
+
+    this.refreshManager_.registerElement(this.element, refresher => {
+      console.log('Eureka!');
+      this.isRefreshing_ = true;
+      this.unlayoutCallback();
+      this.onLayoutMeasure();
+      this.adPromise_.then(() => {
+        if (!this.isRefreshing_) {
```

Comment:
When would what occur, specifically? If you mean this.isRefreshing_ being set to false, then this could happen if we attempt to collapse the slot in onLayoutMeasure.